hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/dispatch/ChildDispatchFactory.java

```diff
@@ -229,66 +224,69 @@ private RecordDispatch newChildDispatch(
             @NonNull final StoreMetricsService storeMetricsService,
             @NonNull final ExchangeRateManager exchangeRateManager,
             @NonNull final TransactionDispatcher dispatcher) {
-        final var readableStoreFactory = new ReadableStoreFactory(stack);
+        final var readableStoreFactory = new ReadableStoreFactory(childStack);
         final var writableStoreFactory = new WritableStoreFactory(
-                stack, serviceScopeLookup.getServiceName(txnInfo.txBody()), config, storeMetricsService);
-        final var serviceApiFactory = new ServiceApiFactory(stack, config, storeMetricsService);
+                childStack, serviceScopeLookup.getServiceName(txnInfo.txBody()), config, storeMetricsService);
+        final var serviceApiFactory = new ServiceApiFactory(childStack, config, storeMetricsService);
+        final var priceCalculator =
+                new ResourcePriceCalculatorImpl(consensusNow, txnInfo, feeManager, readableStoreFactory);
+        final var storeFactory = new StoreFactoryImpl(readableStoreFactory, writableStoreFactory, serviceApiFactory);
+        final var entityNumGenerator = new EntityNumGeneratorImpl(
+                new WritableStoreFactory(childStack, EntityIdService.NAME, config, storeMetricsService)
+                        .getStore(WritableEntityIdStore.class));
+        final var recordBuilders = new RecordBuildersImpl(childStack);
         final var dispatchHandleContext = new DispatchHandleContext(
                 consensusNow,
                 creatorInfo,
                 txnInfo,
                 config,
                 authorizer,
                 blockRecordManager,
-                new ResourcePriceCalculatorImpl(consensusNow, txnInfo, feeManager, readableStoreFactory),
+                priceCalculator,
                 feeManager,
-                new StoreFactoryImpl(readableStoreFactory, writableStoreFactory, serviceApiFactory),
+                storeFactory,
                 payerId,
                 keyVerifier,
                 platformState,
                 topLevelFunction,
                 Key.DEFAULT,
                 exchangeRateManager,
-                stack,
-                new EntityNumGeneratorImpl(
-                        new WritableStoreFactory(stack, EntityIdService.NAME, config, storeMetricsService)
-                                .getStore(WritableEntityIdStore.class)),
+                childStack,
+                entityNumGenerator,
                 dispatcher,
                 recordCache,
                 networkInfo,
-                new RecordBuildersImpl(recordBuilder, recordListBuilder, config),
+                recordBuilders,
                 this,
                 dispatchProcessor,
-                recordListBuilder,
                 throttleAdviser);
+        final var childFees = computeChildFees(dispatchHandleContext, category, dispatcher, topLevelFunction, txnInfo);
```

Comment:
Since Fees are also required to construct a Dispatch [here](https://github.com/hashgraph/hedera-services/blob/7de1e47e2bfa1eafe0ffbc4d9e676100c6a5e02f/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/Dispatch.java#L96),, it is computed here. Where do you recommend to computeFees?